examples/dev_flow/sanitizers/compiler_sanitizers.rst

```diff
@@ -0,0 +1,356 @@
+.. _examples_dev_flow_sanitizers_compiler_sanitizers:
+
+Compiler sanitizers
+===================
+
+.. warning::
+
+   Using sanitizers in production, especially with programs that run with elevated privileges (for example, SUID binaries on Linux), is dangerous.
+   The sanitizer runtime libraries depend on environment variables, which could allow privilege escalation attacks.
+   Use sanitizers only in development and testing environments.
+   runtimes rely on environment variables that could enable privilege escalation attacks.
+   Use sanitizers only in development and testing environments.
+
+Sanitizers are powerful tools for detecting runtime bugs like buffer overflows, data races, memory leaks,
+dangling pointers, use-of-uninitialized memory, and various types of undefined behavior. Compilers such as
+GCC, Clang, and MSVC support these tools through specific compiler and linker flags.
+
+This page explains recommended approaches for integrating compiler sanitizers into your workflow with Conan.
+
+Modeling and applying sanitizers using settings
+-----------------------------------------------
```

Comment:
I think it would be good, before this section, to add a bit of explanation about how sanitizers influence binary compatibility and then, as a result, explain how you can model it for these cases.

Also, it would be great to do some research about if all the sanitizers will make binaries to break compatibility or there are cases were you can apply the sanitizers just as flags without the risk of producing incompatible binaries.